deebot_client/map.py

```diff
@@ -314,74 +428,98 @@ def get_base64_map(self, width: int | None = None) -> bytes:
             and width == self._last_image.width
             and not self._map_data.changed
         ):
-            _LOGGER.debug("[get_base64_map] No need to update")
-            return self._last_image.base64_image
+            _LOGGER.debug("[get_svg_map] No need to update")
+            return self._last_image.svg_image
 
-        _LOGGER.debug("[get_base64_map] Begin")
-        image = Image.new("RGBA", (6400, 6400))
-        draw = DashedImageDraw(image)
+        _LOGGER.debug("[get_svg_map] Begin")
 
+        image = Image.new("RGBA", (6400, 6400))
+        draw = ImageDraw.ImageDraw(image)
         self._draw_map_pieces(draw)
```

Comment:
Unfortunately no. Considering that map pieces come as a sort of bitmap, It would require some sort of raster to vector conversion to get anything printable as raw SVG elements. I don't know if it would worth the effort and, probably, the final result would be almost the same.

Reply:
Ok improving the rest is already a nice job and thank you 🙏